File: perl/Template.pl
#!/usr/bin/perl
package Local::Ajeetmurty::Reference::Perl;

use strict;
use warnings 'FATAL' => 'all';
use Log::Log4perl;

my $LOG_CONF_FILE = "log4perl.conf";
Log::Log4perl::init($LOG_CONF_FILE);
my $logp = Log::Log4perl->get_logger();

sub do_something {
	$logp->info('doing log info');
	$logp->debug('doing log debug');
	$logp->error('doing log error');
}

#main
$logp->info('start');
eval { do_something(); };
if ($@) {
	$logp->error( 'error: ', $@ );
}
$logp->info('stop');

File: java/src/main/java/ajeetmurty/reference/java/Template.java
package ajeetmurty.reference.java;

import java.io.IOException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Template {
	private final Logger logp = LoggerFactory.getLogger(this.getClass().getName());

	public static void main(String[] args) {
		new Template();
	}

	public Template() {
		doSomething();
	}

	private void doSomething() {
		logp.info("start");
		try {
			logp.info("doing info");
			logp.debug("doing debug");
			logp.error("doing error");
			throw new IOException("throw io exception");
		} catch (Exception e) {
			logp.error(e.getMessage(), e);
		}
		logp.info("stop");
	}
}